PostgreSQL already uses -fPIC instead of -fpic on SPARC.  In my test run, I
found that it was needed on s390x as well, so change the test to do this for
either (or rather, to not do this for anything else).

--- postgresql-7.3.4/src/makefiles/Makefile.linux	2003-07-22 18:12:16.000000000 -0400
+++ postgresql-7.3.4/src/makefiles/Makefile.linux	2003-07-22 18:12:04.000000000 -0400
@@ -5,7 +5,7 @@
 allow_nonpic_in_shlib = yes
 DLSUFFIX = .so
 
-ifeq "$(findstring sparc,$(host_cpu))" "sparc"
+ifneq "$(filter sparc s390 s390x,$(host_cpu))" ""
 CFLAGS_SL = -fPIC
 else
 CFLAGS_SL = -fpic
